web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js

```diff
@@ -146,13 +146,15 @@ odoo.define('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer', function (requ
             if (_.isNull(description)) {
                 description = node.attrs.string || field.string;
             }
-            $th.text(description).data('name', name);
+            $th.html(description).data('name', name);
```

Comment:
By doing this, modules using this module should escape HTML manually, and a simple translation could inject a XSS. I think it's a bad idea in general terms. Have you tried writing i.e. your description as `Mon, Feb 15` (that's a real nbsp there between Feb and 15).

Reply:
The security concern is valid, yet by not-trusting a module or it's translation (which is a resource for system) within the system is kind-a odd

Reply:
@alexey-pelykh I'm afraid you can't trust anything...

Reply:
Anyways the problem of somebody having a header such as `Count < total` becoming suddenly broken unles you change it to `Count &lt; total` is still valid. IMHO we gain much more by autoescaping than by allowing to use nbsp.

Reply:
I guess @Yajo is right, we'd gain more issues than benefits for it. I'll try out the unicode nbsp symbol w/o html and see if it will work, otherwise I think this PR would get transformed to `html(escape(input))` in order to translate unicode nbsp to `&nbsp;` somehow

Reply:
That seems a better idea, yes. In any case, before continuing I think the best is that you test if just pasting a raw nbsp character works for you, and in such case this can be closed.